### DISCUSSION TITLE
WixDotNetCoreBootstrapperApplicationHost

### DISCUSSION THREAD

**Discussion by Steve0212a** [wixtoolset/issues#8872] [Questions]
I am in the process of upgrading our installer MSI and bootstrapper from WIX v4 to v5.  For the bootstrapper, we were using WixDotNetCoreBootstrapperApplicationHost as we have a custom .Net 8 self-contained core UI.   WixDotNetCoreBootstrapperApplicationHost is now deprecated in v5.  The docs say to use WixPrerequisiteBootstrapperApplication.

However, when I use that, I have to set at least one pre-req to bal:PrereqPackage="yes".  When I do this, that prereq is automatically installed before the install starts.

My issue is that all of my prereqs are dependent on what the user chooses to install.  For example, if the user chooses to install option 1, I need prereq 1.  If the user chooses to install option 2, I need prereq 2.  This worked without issue with WixDotNetCoreBootstrapperApplicationHost .  Our software is very complex with multiple components.  Depending on which components is selected to be installed, different prereqs are needed.  There is no one prereq that is always needed.

For now, I have set the simplest prereq to be the bal:PrereqPackage="yes" as a work around.  

Is there any way to not install any prereqs with a custom .Net 8 self-contained core UI where that UI sets what to install?

Thanks

**Comment by nirbar** [comment]
The meaning of prerequisites is any package that the UI needs to be installed before it can load
You should publish the BA as a self-contained and use the publish folder as the BA folder
Then you'll have no prerequisites at all

**Reply by Steve0212a** [reply]
that is what I am doing.  However, the docs say to use WixPrerequisiteBootstrapperApplication because WixDotNetCoreBootstrapperApplicationHost is deprecated which requires at least one prereq.  What application should I use if I have no prereqs since WixDotNetCoreBootstrapperApplicationHost is deprecated?

**Comment by nirbar** [comment]
You need to use `BootstrapperApplication`, not `WixPrerequisiteBootstrapperApplication` 
~~~
    <BootstrapperApplication SourceFile="SampleBA.exe" >
      <PayloadGroupRef Id="SampleBA"/>
    </BootstrapperApplication>
~~~

**Reply by Steve0212a** [reply]
I misunderstood.  I thought one of the bal schema objects was required inside the BootstrapperApplication.  Thanks for clarifying.
